var/spack/repos/builtin/packages/zlib-ng/package.py

```diff
@@ -76,11 +76,12 @@ def libs(self):
     def flag_handler(self, name, flags):
         if name == "cflags" and self.spec.satisfies("+pic build_system=autotools"):
             flags.append(self.compiler.cc_pic_flag)
+        if name == "ldflags" and self.spec.satisfies("%cce@17"):
+            flags.append("-Wl,--undefined-version")
```

Comment:
Why?!

Reply:
See the explanation here: https://github.com/spack/spack/pull/44010#discussion_r1628365884. Other compilers using LLD can be affected as well but I'm not sure, so I kept the compiler check to what I've tested.